checker/src/org/checkerframework/checker/lock/LockVisitor.java

```diff
@@ -354,9 +606,342 @@ protected boolean skipContractCheck(Tree tree, FlowExpressions.Receiver expr, Fl
                 }
             }
         } catch (FlowExpressionParseException e) {
-            // errors are reported at declaration site
+            checker.report(e.getResult(), tree);
         }
 
         return false;
     }
+
+    @Override
+    public boolean isValidUse(AnnotatedDeclaredType declarationType,
+            AnnotatedDeclaredType useType, Tree tree) {
+        declarationType.replaceAnnotation(GUARDEDBY);
```

Comment:
This is called `by BaseTypeValidator.visitDeclared`. Given a class such as:

    static class MyClass { public Object field; }
when validating a declaration of an instance of this class such as:

    @GuardedBy("this") MyClass foo = new MyClass();
`visitDeclared` will ensure that the declared type of `foo` is a subtype of the type for `MyClass` as returned by `annotateImplicit`, which is `@GuardedBy({})`. Since `@GuardedBy("this")` is not a subtype of `@GuardedBy({})`, this check would fail, hence the replacement of annotations with `@GuardedBy({})`.

Note that if I change the declaration of `MyClass` to:

    static @GuardedByUnknown class MyClass { public Object field; }
then the calls to `replaceAnnotation` in this method become unnecessary when validating the declaration of variable `foo`.